app/models/miq_server/role_management.rb

```diff
@@ -50,10 +50,15 @@ def set_active_role_flags
     save
   end
 
-  def set_assigned_roles
+  def sync_assigned_roles
     self.role = ::Settings.server.role
   end
 
+  def ensure_default_roles
+    MiqServer.my_server.add_settings_for_resource(:server => {:role => ENV["MIQ_SERVER_DEFAULT_ROLES"]}) if role.blank? && ENV["MIQ_SERVER_DEFAULT_ROLES"].present?
```

Comment:
I'm wondering if it's cleaner to extract this into a separate method like maybe `save_settings` and call it directly...

https://github.com/ManageIQ/manageiq/blob/a13b67bb6d508c1118bf65661cb4d41d14470725/app/models/mixins/configuration_management_mixin.rb#L13-L16